containers/jersey2-routing/src/main/java/com/yahoo/gondola/container/RoutingFilter.java

```diff
@@ -0,0 +1,234 @@
+package com.yahoo.gondola.container;
+
+import com.yahoo.gondola.Cluster;
+import com.yahoo.gondola.Config;
+import com.yahoo.gondola.Gondola;
+import com.yahoo.gondola.Member;
+
+import org.apache.http.Header;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.InputStreamEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.protocol.HTTP;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import javax.servlet.ServletException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Response;
+
+public class RoutingFilter implements ContainerRequestFilter {
+
+    public static final String X_GONDOLA_LEADER_ADDRESS = "X-Gondola-Leader-Address";
+    public static final String APP_PORT = "appPort";
+    public static final String APP_SCHEME = "appScheme";
+    /**
+     * Routing table
+     * Key: memberId, value: HTTP URL
+     */
+    ClusterIdCallback clusterIdCallback;
+
+    Gondola gondola;
+    Set<String> myClusterIds;
+
+    // clusterId --> list of available servers
+    Map<String, List<String>> routingTable;
+
+    CloseableHttpClient httpclient;
+    public RoutingFilter(Gondola gondola, ClusterIdCallback clusterIdCallback) throws ServletException {
```

Comment:
In routing filter, it needs following function from gondola, which is dynamic.
- Gondola.getLeader()  - This is mandatory to pass the gondola class to it.
- Listening to gondola event, to update internal routing table 
- Get active config of gondola instance.